modules/activiti-engine/src/main/java/org/activiti/engine/impl/history/DefaultHistoryManager.java

```diff
@@ -407,8 +409,12 @@ public void recordExecutionReplacedBy(ExecutionEntity execution, InterpretableEx
         .executionId(execution.getId())
         .unfinished()
         .list();
+      
+      // keep record of unfinished tasks to only update related variables later
+      Set<String> unfinishedTasks = new HashSet<String>();
       for (HistoricTaskInstanceEntity historicTaskInstance: historicTaskInstances) {
         historicTaskInstance.setExecutionId(replacedBy.getId());
+        unfinishedTasks.add(historicTaskInstance.getId());
```

Comment:
We need to consider unfinished entities as well, so we need to add the cached entities to unfinishedTasks. Specifically, it is the following part.

https://github.com/Activiti/Activiti/blob/a43ff77ed55f881abeb7da87125b29f0bbd30df6/modules/activiti-engine/src/main/java/org/activiti/engine/impl/history/DefaultHistoryManager.java#L403

Reply:
do you have a case when this is needed?

Reply:
@balsarori 
I am sorry to have kept you waiting.

I added the case when this is need in the following pull request.
https://github.com/Activiti/Activiti/pull/1739